llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -82,6 +90,15 @@ AIE2TargetMachine::AIE2TargetMachine(const Target &T, const Triple &TT,
   setGlobalISelAbort(GlobalISelAbortMode::Enable);
 }
 
+/// Predicate for Internalize pass.
+static bool mustPreserveGV(const GlobalValue &GV) {
+  if (const Function *F = dyn_cast<Function>(&GV))
+    return F->isDeclaration() || F->getName().starts_with("_main");
```

Comment:
I just pushed a PR to change this to "main" in the startup code. I don't know the rationale for _main, nor did @konstantinschwarz. Everything would look more familiar if we could just use main, like the rest of the world.

Reply:
Hi, maybe we can have a option like this for flexibility:
```
cl::opt<std::string> EntryFunctionName ...
```

Reply:
Having an option for this sounds good! It could default to `main`, but would allow users to specify their own entry function

Reply:
Instead of `.starts_with`, change to `equals`? Otherwise we might get false positives

Reply:
Instead of .starts_with, change to equals? Otherwise we might get false positives

Reply:
Made the changes.

Reply:
I think its sensible for the `FunctionSkipList` to contain `main` by default. Now we require both options to be set